packager.go

```diff
@@ -291,11 +320,23 @@ func (this Packager) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	fetchReq, fetchResp, httpErr := this.fetchURL(fetchURL)
+	fetchReq, fetchResp, httpErr := this.fetchURL(fetchURL, req.Header)
 	if httpErr != nil {
 		httpErr.LogAndRespond(resp)
 		return
 	}
+
+	// If fetchURL returns a 304, then also return a 304 with appropriate headers.
+	if fetchResp.StatusCode == 304 {
+		resp.WriteHeader(http.StatusNotModified)
```

Comment:
If my read of https://golang.org/pkg/net/http/#ResponseWriter is correct, then any headers you set after a call to `WriteHeader()` are ignored. I think you need to move this line after the for loop?

Also, I'm not positive, but I think you need to call `resp.Write([]byte{""})` in order to close the connection.